### PR TITLE
Update participants presentation in group details view controller

### PR DESCRIPTION
**Pull request by daehn**
## What's new in this PR?

The participants presentation in group details view controller has been updated and now follows the following rules:

* We limit the amount of group participants we show to 7.
* If the amount of participants exceeds 7, we only show the first 5 and a "Show All" cell.
* The "Show All" cell opens a new view controller showing all participants and a search bar.

### CODE REVIEW

File: Wire-iOS Tests/GroupParticipantsDetailViewControllerTests.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
+//
+// Wire
+// Copyright (C) 2016 Wire Swiss GmbH
```
Comment (typfel) [inline Wire-iOS Tests/GroupParticipantsDetailViewControllerTests.swift]: 2016 > 2018

**Comment by daehn** [comment]
@typfel updated, also added a dismiss button to the details screen and adjusted the snapshot test to include a navigation bar

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+fileprivate extension String {
+    var isValidQuery: Bool {
+        return !isEmpty && self != "@"
+    }
+}
+
+class GroupParticipantsDetailViewModel: NSObject, SearchHeaderViewControllerDelegate {
+
+    private let __participants: [ZMBareUser]
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewModel.swift]: @daehn not sure if we have anywhere the double (or single) underscore prefix, what is the meaning of it?

**Comment by mikeger** [comment]
We have the very similar UI in `ReactionsListViewController`, may be worth unifying it?

**Comment by daehn** [comment]
@mikeger AFAIK design has some plans for `ReactionsListViewController` as well, so I am not sure if would make sense right now. But I'll check with them.

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewModel.swift (hunk 2 above)
Comment (daehn) [reply Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewModel.swift]: Good point I think we use one underscore in some places, but maybe better to just prefix it with `internal` instead.

Comment (daehn) [reply Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewModel.swift]: @mikeger done in https://github.com/wireapp/wire-ios/pull/2388

**Pull request merged**